src/mlpack/tests/feedforward_network_test.cpp

```diff
@@ -621,4 +623,94 @@ BOOST_AUTO_TEST_CASE(OptimizerTest)
   model.Train(trainData, trainLabels, opt);
 }
 
+/**
+ * Train the RBF network on a larger dataset.
+ */
+BOOST_AUTO_TEST_CASE(RBFNetworkTest)
+{
+  // Load the dataset.
+  arma::mat trainData;
+  data::Load("thyroid_train.csv", trainData, true);
+
+  arma::mat trainLabels = trainData.row(trainData.n_rows - 1);
+  trainData.shed_row(trainData.n_rows - 1);
+
+  arma::mat testData;
+  data::Load("thyroid_test.csv", testData, true);
+
+  arma::mat testLabels = testData.row(testData.n_rows - 1);
+  testData.shed_row(testData.n_rows - 1);
+
+  /*
+   * Construct a feed forward network with trainData.n_rows input nodes,
+   * hiddenLayerSize hidden nodes and trainLabels.n_rows output nodes. The
+   * network structure looks like:
+   *
+   *  Input         RBF          Activation    Output
+   *  Layer         Layer         Layer        Layer
+   * +-----+       +-----+       +-----+       +-----+
+   * |     |       |     |       |     |       |     |
+   * |     +------>|     +------>|     +------>|     |
+   * |     |       |     |       |     |       |     |
+   * +-----+       +--+--+       +-----+       +-----+
+   */
+
+  arma::mat centroids;
+  KMeans<> kmeans;
+  kmeans.Cluster(trainData, 8, centroids);
+
+  FFN<NegativeLogLikelihood<> > model;
+  model.Add<RBF<> >(trainData.n_rows, 8, centroids);
+  model.Add<Linear<> >(8, 3);
+  model.Add<LogSoftMax<> >();
+
+  TestNetwork<>(model, trainData, trainLabels, testData, testLabels, 10, 0.1);
+
+  arma::mat dataset;
+  dataset.load("mnist_first250_training_4s_and_9s.arm");
+
+  // Normalize each point since these are images.
+  for (size_t i = 0; i < dataset.n_cols; ++i)
+  {
+    dataset.col(i) /= norm(dataset.col(i), 2);
+  }
+
+  arma::mat labels = arma::zeros(1, dataset.n_cols);
+  labels.submat(0, labels.n_cols / 2, 0, labels.n_cols - 1).fill(1);
+  arma::mat labels1 = arma::zeros(2, dataset.n_cols);
+  for(size_t i = 0; i < dataset.n_cols; i++)
+  {
+    labels1.col(i).row(labels(i)) = 1;
+  }
+  labels += 1;
```

Comment:
what are the values of labels for the predicted class and the other class?

Reply:
The values of labels should be 1 for target class and 0 for all the other classes.

Reply:
Yes inside labels1 I am using 1 for Target class 0 for all other class. Same as you suggested.

Reply:
alright thanks for the clarification